### DISCUSSION TITLE
Cache purge

### DISCUSSION THREAD

**Discussion by schack** [honojs/hono#788] [General]
Hi 
How can I purge a cached URL I have cached using the middleware function?

**Comment by yusukebe** [comment]
Hi @schack !

On Cloudflare, Currently, we have to access the API Endpoints to purge the cache.

<https://developers.cloudflare.com/cache/how-to/purge-cache/>

I only know this way to purge caches. I just hope there's a smarter way to do it.

**Reply by schack** [reply]
OK, that can work, but how do I figure out the key name to purge, is it somehow constructed based on the cache name and the URL ?

**Reply by yusukebe** [reply]
I do purge by URL.

The following is the code I wrote before. Create middleware for purge:

```ts
const purgeMiddleware: Handler = async (c, next) => {
  const raceId = c.req.param("raceId");
  const url = new URL(c.req.url);

  const apiURL = `https://api.cloudflare.com/client/v4/zones/${c.env.ZONE_ID}/purge_cache`;
  const data = {
    files: [`https://${url.hostname}/api/races/${raceId}`],
  };

  const fetchResponse = await fetch(apiURL, {
    body: JSON.stringify(data),
    headers: {
      Authorization: `Bearer ${c.env.API_TOKEN}`,
      "Content-Type": "application/json",
    },
    method: "POST",
  });
  console.log(await fetchResponse.json());
  await next();
};
```

And use it:

```ts
app.post("/api/races/:raceId/betting-tickets", purgeMiddleware, passHandler);
app.post("/api/initialize", passHandler);
```

If you are in an enterprise plan, it is easier to purge because you can do it by "prefix".
